build-system/eslint-rules/camelcase.js

```diff
@@ -46,6 +46,17 @@ module.exports = function (context) {
         }
       }
 
+      // Permit object destructuring, since that is similar to membership access.
+      // Requires that the key is immediately renamed to a conforming value.
+      if (
+        parent.type === 'Property' &&
+        parent.parent.type === 'ObjectPattern' &&
+        parent.key?.name !== parent.value?.name &&
+        !parent.value?.name.includes('_')
```

Comment:
```suggestion
        parent.key === node
```

Reply:
I think this only covers the first condition. We also want to ensure the renamed identifier doesn't have underscores.

Reply:
If we're checking for camelcase, why exclude `'_'` rather than test with a regex like `/[a-zA-Z]+/`?

Reply:
> I think this only covers the first condition (but also needs a negation). We also want to ensure the renamed identifier doesn't have underscores.

The intention here is to allow the key to be access, but the value to throw. This condition checks if we're the key, and early returns. [Repl](https://astexplorer.net/#/gist/c2eec1ad4d7ba4229aa6f1b44a14f350/7a4ff9c37e3d844f086d0d1c0b700351d4a8c755)

> If we're checking for camelcase, why exclude `'_'` rather than test with a regex like `/[a-z][a-z0-9]+/i`?

Searching for a single char is faster than performing a regex test.

Reply:
Ah, that makes sense that each Ident gets its own call. Comitting.